backend/common/kernel.py

```diff
@@ -22,126 +27,98 @@
 from backend.common.utils import csv_to_json
 from backend.common.ai_drive import dl_tabular_drive, ml_drive, dl_img_drive
 from backend.dl.detection import detection_img_drive
-from backend.aws_helpers.dynamo_db_utils.execution_db import updateStatus
 
 
 def router(msg):
     """
     Routes the message to the appropriate training function.
     """
     print("Message received")
-    request_data = json.loads(msg)
-    print(f"{execution_id} is marked as STARTING")
-    entryData = {
-        "execution_id": msg["execution_id"],
-        "user_id": msg["user"]["uid"],
-        "name": msg["custom_model_name"],
-        "data_source": msg["data_source"],
-        "status": "STARTING",
-        "timestamp": get_current_timestamp(),
-        "progress": 0,
-    }
-    updateStatus(execution_id=execution_id, status="STARTING", entryData=entryData)
-    if msg["route"] == "tabular-run":
-        result = tabular_run_route(msg)
+    request_data = TrainspaceData(**(json.loads(msg)))
+    trainspace_id = request_data.trainspace_id
+    print(f"{trainspace_id} is marked as STARTING")
+    updateStatus(trainspace_id, TrainStatus.STARTING.name)
+    data_source = request_data.data_source
+
+    if data_source == "TABULAR":
```

Comment:
what is stopping us from delegating `data_source == "TABULAR"` to say the `tabular_helper()` function, the `data_source == "IMAGE" to say the `image_helper()` function, etc. This would make the `router(msg)` much more maintainable and readable?